modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js

```diff
@@ -17,244 +17,302 @@
+    $("#y_axis-temperature").html("");
+    $("#smoother-temperature").html("");
+    $("#legend-temperature").html("");
+    $("#chart-temperature").html("");
+    $("#x_axis-temperature").html("");
+    $("#slider-temperature").html("");
+
+    $("#y_axis-coffeelevel").html("");
+    $("#smoother-coffeelevel").html("");
+    $("#legend-coffeelevel").html("");
+    $("#chart-coffeelevel").html("");
+    $("#x_axis-coffeelevel").html("");
+    $("#slider-coffeelevel").html("");
+
     var devices = $("#connectedcup-details").data("devices");
     var tzOffset = new Date().getTimezoneOffset() * 60;
 
-    var streamIndex = 0;
-    var streams = ["temperature", "coffeelevel"];
+    var chartWrapperElmId = "#connectedcup-div-chart";
+    var graphWidth = $(chartWrapperElmId).width() - 50;
+    var temperatureGraphConfig = {
+        element: document.getElementById("chart-temperature"),
+        width: graphWidth,
+        height: 400,
+        strokeWidth: 2,
+        renderer: 'line',
+        interpolation: "linear",
+        unstack: true,
+        stack: false,
+        xScale: d3.time.scale(),
+        padding: {top: 0.2, left: 0.02, right: 0.02, bottom: 0.2},
+        series: []
+    };
 
-    populateGraph();
+    var coffeelevelGraphConfig = {
+        element: document.getElementById("chart-coffeelevel"),
+        width: graphWidth,
+        height: 400,
+        strokeWidth: 2,
+        renderer: 'line',
+        interpolation: "linear",
+        unstack: true,
+        stack: false,
+        xScale: d3.time.scale(),
+        padding: {top: 0.2, left: 0.02, right: 0.02, bottom: 0.2},
+        series: []
+    };
 
-    function populateGraph() {
-        if (streamIndex < 2) {
-            retrieveDataAndDrawLineGraph(streams[streamIndex], from, to);
+    if (devices) {
+        for (var i = 0; i < devices.length; i++) {
+            temperatureGraphConfig['series'].push(
+                    {
+                        'color': palette.color(),
+                        'data': [{
+                            x: parseInt(new Date().getTime() / 1000),
+                            y: 0
+                        }],
+                        'name': devices[i].name
+                    });
+
+            coffeelevelGraphConfig['series'].push(
+                {
+                    'color': palette.color(),
+                    'data': [{
+                        x: parseInt(new Date().getTime() / 1000),
+                        y: 0
+                    }],
+                    'name': devices[i].name
+                });
         }
-        streamIndex++;
+    } else {
+        temperatureGraphConfig['series'].push(
+                {
+                    'color': palette.color(),
+                    'data': [{
+                        x: parseInt(new Date().getTime() / 1000),
+                        y: 0
+                    }],
+                    'name': $("#connectedcup-details").data("devicename")
```

Comment:
fixed it with https://github.com/wso2/product-iots/pull/190/commits/418ed9baae8a91bc0384ef6f1fcd18205b969f3d